xptifw/samples/sycl_perf_collector/sycl_perf_collector.cpp

```diff
@@ -0,0 +1,891 @@
+    trace_origin = "thread_end";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::wait_begin:
+    trace_origin = "wait_begin";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::wait_end:
+    trace_origin = "wait_end";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::function_begin:
+    trace_origin = "function_begin";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::function_end:
+    trace_origin = "function_end";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::metadata:
+    trace_origin = "metadata";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::function_with_args_begin:
+    trace_origin = "function_with_args_begin";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::function_with_args_end:
+    trace_origin = "function_with_args_end";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::mem_alloc_begin:
+    trace_origin = "mem_alloc_begin";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::mem_alloc_end:
+    trace_origin = "mem_alloc_end";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::mem_release_begin:
+    trace_origin = "mem_release_begin";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::mem_release_end:
+    trace_origin = "mem_release_end";
+    break;
+  case (
+      uint16_t)xpti::trace_point_type_t::offload_alloc_memory_object_construct:
+    trace_origin = "offload_alloc_construct";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::offload_alloc_memory_object_destruct:
+    trace_origin = "offload_alloc_destruct";
+    break;
+  case (
+      uint16_t)xpti::trace_point_type_t::offload_alloc_memory_object_associate:
+    trace_origin = "offload_alloc_associate";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::offload_alloc_memory_object_release:
+    trace_origin = "offload_alloc_release";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::offload_alloc_accessor:
+    trace_origin = "offload_alloc_accessor";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::queue_create:
+    trace_origin = "queue_create";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::queue_destroy:
+    trace_origin = "queue_destroy";
+    break;
+  case (uint16_t)xpti::trace_point_type_t::diagnostics:
+    trace_origin = "diagnostics";
+    break;
+  default:
+    trace_origin = "Unknown";
+    break;
+  }
+
+  std::cout << "Record[" << count++ << "] -> " << trace_origin << " details("
+            << r.TSBegin << "," << r.CorrID << "," << r.Name << ")\n";
+}
+
+void record_and_save(const char *StreamName, xpti::trace_event_data_t *Event,
+                     uint16_t TraceType, uint64_t Instance,
+                     const void *UserData) {
+  if (GRecordsInProgress) {
+
+    // For `function_begin` trace point type, the Parent and Event parameters
+    // can be null. However, the `UserData` field must be present and contain
+    // the function name that these trace points are defined to trace.
+    MeasureHandlers timer;
+
+    char *Name;
```

Comment:
I see it as output parameter for xptiRegisterString. Is it feasible to initialize it to nullptr here since we didn;t check xptiRegisterString result and could not handle failure properly otherwise?

Reply:
Done!